addon/mixins/prop-types.js

```diff
@@ -60,7 +60,7 @@ export default Ember.Mixin.create({
       const presentPropKeys = Object.keys(this)
       const defaultProps = this.getDefaultProps()
 
-      const needsDefaultProp = Object.assign({}, ...(Object.keys(defaultProps).forEach((key) => {
+      const needsDefaultProp = Object.assign({}, ...(Object.keys(defaultProps).map((key) => {
```

Comment:
I would also love to see a test that exercises the above bug/fix so it doesn't regress in the future.